### PR TITLE
Parental Leave - Review screen (part 1)

### PR DESCRIPTION
**Pull request by madebynoam**
## What

1. Adds a review screen for the various parental leave application states.
2. Core: If a section name is '', then don't show it in the FormStepper (this allows us to hide sections, without too much impact).

## Screenshots / Gifs

image

## Remaining (for next PR)
1. Connect the review screen action buttons ("View application" and "Edit application")
2. Update the `mapUserToRole` function to work for the different roles

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import { Application, formatText } from '@island.is/application/core'
+import { Box, Icon, Tag, Text } from '@island.is/island-ui/core'
+
+import * as styles from './ReviewSection.treat'
+
+export type reviewSectionState =
+  | 'In progress'
+  | 'Requires action'
+  | 'Complete'
+  | undefined
```
Comment (arnifreyrs) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx:14]: should this be translatable or is this just used as a key? If so, then this should probably be an enum

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    employerRequiresAction: 'Requires action',
+    vinnumalastofnunApproval: 'Complete',
+  },
+  vinnumalastofnun: {
+    vinnumalastofnunApproval: 'In progress',
+    vinnumalastofnunRequiresAction: 'Requires action',
+    approved: 'Complete',
+  },
+}
+
```
Comment (arnifreyrs) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:34]: clever

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  return (
+    <Box marginBottom={10}>
+      <Box>
+        <Text as="div">
+          {formatMessage(mm.reviewScreen.desc)}{' '}
+          <Box display="inlineBlock" marginLeft={1} marginRight={2}>
+            <Button
+              colorScheme="default"
+              iconType="filled"
+              // TODO: Add onClick in next PR
```
Comment (arnifreyrs) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:47]: you basically need to have the Review custom component there rendered behind a state change on click here.

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          <Box display="inlineBlock">
+            <Button
+              colorScheme="default"
+              iconType="filled"
+              // TODO: Add onClick in next PR
+              size="small"
+              type="button"
+              variant="text"
+            >
+              {formatMessage(mm.reviewScreen.buttonsEdit)}
```
Comment (arnifreyrs) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:64]: this one is a bit more tricky to implement. You need to use the submitApplication graphql function

**Review by arnifreyrs — APPROVED** [review]
Great stuff, just minor comments and questions

File: libs/application/ui-shell/src/components/Screen.tsx [hunk 5]
```diff
@@ -193,7 +193,8 @@ const Screen: FC<ScreenProps> = ({
           span={['12/12', '12/12', '7/9', '7/9']}
           offset={['0', '0', '1/9']}
         >
-          <Text variant="h2" marginBottom={5}>
+          {/* <Text variant="h2" marginBottom={5}> */}
```
[the remark this answers is not in the document]
Comment (madebynoam) [reply libs/application/ui-shell/src/components/Screen.tsx]: Good catch thanks, should be fixed now

File: libs/application/templates/parental-leave/src/lib/messages.ts [hunk 6]
```diff
@@ -737,4 +737,66 @@ export const mm: MessageDir = {
...
+    },
+    otherParentTitle: {
+      id: 'pl.application:review.otherParent.title',
+      defaultMessage: 'Other parent approves extra time',
+      description: 'Other parent approves extra time',
+    },
+    otherParentDesc: {
+      id: 'pl.application:review.otherParent.description',
+      defaultMessage:
+        'The other parent will need to approve the extra 14 days you’ve requested.',
```
[the remark this answers is not in the document]
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/lib/messages.ts]: Good catch, was going to make it dynamic, but will leave as generic for now. Should be fixed now.

File: libs/application/templates/parental-leave/src/forms/InReview.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  children: [
+    buildSection({
+      id: 'review',
+      name: '',
+      children: [
+        buildCustomField({
+          id: 'InReviewSteps',
+          // TODO: Can we make the name conditional?
+          // ex: name: state==='approved' ? "Your application is approved" : "Your application is in review"
+          name: mm.reviewScreen.titleInReview,
```
[the remark this answers is not in the document]
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/forms/InReview.ts]: Nice thanks! Should be switched now.

**Review by arnifreyrs — DISMISSED** [review]

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx (hunk 1 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx]: Yeah it's used just as a key, switch to an enum now (great suggestion thanks!).

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx (hunk 3 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:47]: Sounds good, I'll work on this now for next PR.

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx (hunk 4 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:64]: Ooh, sounds good, I'll look into this one next.

**Comment by madebynoam** [comment]
Thanks for the suggestions! Should be fixed now (excluding the two action links) I'll work on those for the next PR next.
@arnifreyrs if you could re-review? Takk!

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
+import React, { FC } from 'react'
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx:1]: Won't this component be also used in others application systems and should be moved to some shared lib?

File: libs/island-ui/core/src/lib/FormStepper/FormStepper.tsx [hunk 9]
```diff
@@ -60,20 +60,26 @@ export const FormStepper: FC<{
...
-            subSection={subSection}
-            isActive={index === activeSection}
-            isComplete={index < activeSection}
-            isLastSection={index === sections.length - 1}
-            sectionIndex={index}
-            activeSubSection={activeSubSection}
-            showSubSectionIcon={showSubSectionIcons}
-          />
-        ))}
+        {sections.map((section, index) => {
```
Comment (jeremybarbet) [inline libs/island-ui/core/src/lib/FormStepper/FormStepper.tsx:63]: You could also do:

```suggestion
        {sections.filter(section => section?.name !== '').map((section, index) => {
```

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx (hunk 8 above)
Comment (madebynoam) [reply libs/application/templates/parental-leave/src/fields/InReviewSteps/ReviewSection/index.tsx:1]: Yes post beta we will discuss this with the design system team and the other application teams. But for now we wanted to do less impact and just add it to the Parental Leave project. 

To make it more re-usable across projects we'll have to make some updates to the state machine and how/if it knows about completed states.

File: libs/island-ui/core/src/lib/FormStepper/FormStepper.tsx (hunk 9 above)
Comment (madebynoam) [reply libs/island-ui/core/src/lib/FormStepper/FormStepper.tsx]: Nice, I like it thanks. Should be updated now.

**Comment by madebynoam** [comment]
@arnifreyrs @jeremybarbet thanks for the comments! Should be up to date now.

**Review by arnifreyrs — APPROVED** [review]

**Review by jeremybarbet — APPROVED** [review]

**Pull request merged**